syllabus/examples/experimental/sb3_procgen_plr.py

```diff
@@ -47,70 +213,149 @@ def __init__(self, curriculum, verbose=0):
         self.curriculum = curriculum
 
     def _on_step(self) -> bool:
-        tasks = self.training_env.venv.venv.venv.get_attr("task")
-
-        update = {
-            "update_type": "on_demand",
-            "metrics": {
-                "value": self.locals["values"],
-                "next_value": self.locals["values"],
-                "rew": self.locals["rewards"],
-                "dones": self.locals["dones"],
-                "tasks": tasks,
-            },
-        }
-        self.curriculum.update_curriculum(update)
+        if args.curriculum and args.curriculum_method == "plr":
+            tasks = self.training_env.venv.venv.venv.get_attr("task")
+            update = {
+                "update_type": "on_demand",
+                "metrics": {
+                    "value": self.locals["values"],
+                    "next_value": self.locals["values"],
+                    "rew": self.locals["rewards"],
+                    "dones": self.locals["dones"],
+                    "tasks": tasks,
+                },
+            }
+            self.curriculum.update_curriculum(update)
         return True
+    
+class Level_Replay_EvalCallback(BaseCallback):
+
+    def __init__(self, env_name, model, eval_freq, num_episodes, num_levels=0, verbose=0):
+        super().__init__(verbose)
+        self.env_name = env_name
+        self.model = model
+        self.num_episodes = num_episodes
+        self.num_levels = num_levels
+        self.eval_freq = eval_freq
 
+    def _on_step(self) -> bool:
+        """
+        This method will be called by the model after each call to `env.step()`.
+        You must return True to continue training, or False to stop.
+        """
+        return True  
+
+    def _on_rollout_end(self) -> None:
+        if self.num_timesteps % self.eval_freq == 0:
```

Comment:
This isn't necessary, we should just evaluate every time this function is called. It should happen every 16,000 steps, but try running it and make sure that's what happens